File: training/javascript/src/de/sots/cellarsandcentaurs/application/encounter.mjs
import { CreatureFacade } from "./index.js";
import { rollD20 } from "../domain/model/Dice.js";
import { Centaur } from "../domain/model/Centaur.js";

export class EncounterRunner {
    constructor(creatureFacade) {
        this.creatureFacade = creatureFacade;
    }

    start(creatures) {
        const stable = CreatureFacade.STABLE_NAME;
        const initiative = creatures.map((creature) => [rollD20(), creature]);
        initiative.sort((left, right) => right[0] - left[0]);
        return initiative.map(([, creature]) => (creature instanceof Centaur ? creature.gallop() : 0));
    }
}

File: training/python/de/sots/cellarsandcentaurs/application/encounter.py
try:
    from de.sots.cellarsandcentaurs.domain.model.dice import roll_d20
except ImportError:
    from de.sots.cellarsandcentaurs.domain.model.fallback_dice import roll_d20


class Encounter:
    def surprise_round(self) -> bool:
        return roll_d20().value > 10
